### ISSUE TITLE
Platform not found: light.lightwaverf_light

### ISSUE THREAD

**Issue by RevillWeb** [SCCMOG/Home-Assistant#2]
Hey,

I've got it all setup per your instructions in the README however when I validate by config I get the following error:

`Platform not found: light.lightwaverf_light`

I'm sure I've just done something wrong, any help would be greatly appreciated!

Thank you.

**Comment by RevillWeb** [comment]
Looks like its failing to import `pika` in lightwaverd_light.py. `pika` appears to be importing fine in `LightwaveRF_service.py` so not sure whats going on.

**Comment by SCCMOG** [comment]
Hi Revil, jump into the virtual environment for Home assistant and install Pika. It should load this automatically but it seems to be failing for you.

`$ sudo su -s /bin/bash homeassistant`
`$ source /srv/homeassistant/bin/activate`
`$ pip3 install pika`
`$ exit`
`$ sudo reboot`

That should do the trick :)

Cheers,

Richie

**Comment by SCCMOG** [comment]
I have just updated the component. Try that version :)

**Comment by RevillWeb** [comment]
Hey, thank you! I'll give it a go when I get home this evening 👍

**Comment by RevillWeb** [comment]
So when I try `sudo su -s /bin/bash homeassistant` I get `No passwd entry for user 'homeassistant'`. I'm on Ubuntu not a PI if that makes any difference. Thank you for your help.

**Comment by SCCMOG** [comment]
Ahhh ok so you have installed the all in one without a virtual environment? Or how did you install Hass?

**Comment by RevillWeb** [comment]
I followed these instructions: https://www.home-assistant.io/docs/installation/virtualenv/ :)

**Comment by SCCMOG** [comment]
Use step 2 and 3 then you will be in the virtual environment, then run the pika install while in the virtual environment :)

**Comment by RevillWeb** [comment]
Brilliant! That worked. I have the switches for two lights and can see the messages going into RabbitMQ but they are not coming on. I'll investigate to see what else I may have done wrong. Is there any log files I should look at? Thanks again for your continued support, very much appreciated.

**Comment by SCCMOG** [comment]
Have you paired your raspberry pi with the WiFi link?

**Comment by RevillWeb** [comment]
Yep. I currently have OpenHab setup and working. I'll spend some time looking at the packets sent over the network to make sure they are the same. Perhaps I configured something wrong. Thank you.

**Comment by SCCMOG** [comment]
Is your WiFi link on the same subnet?

**Comment by RevillWeb** [comment]
Ah, so I'm testing all this in a VM on a Windows 10 PC with NAT (before I roll it to replace OpenHab on my Ubuntu Server). I know I can communicate with the Wifi Link because I can ping it for one and when I send the pair command it flashes - but is there something else about the subnet which might cause issues?

**Comment by SCCMOG** [comment]
Ok so you have clicked the pair button on the WiFi link when it flashes yes?

Yea my script uses UDP to the broadcast address, 255.255.255.255. You could change the IP in the script to point directly at your link instead. If that works I will change the code to have a default of the 255 address but an overdue in the yaml entry for a direct IP of the link :)

**Comment by RevillWeb** [comment]
I'll give that a go, I have historically always used the specific IP and NOT the broadcast address. I'll let you know when I'm home 👍 Thanks again!

**Comment by RevillWeb** [comment]
So, that doesn't appear to make a difference. Should the messages in the rabbit queue stay there forever?

**Comment by SCCMOG** [comment]
No mate they should be consumed ;) check you rabbit mq config :)

**Comment by RevillWeb** [comment]
In the README it says to create a queue and exchange named `LightwaveRf` but looking at the code it needs an uppercase F too: `LightwaveRF`? I had already done this for the queue. Just tried that with the exchange as well but that doesn't appear to change anything either. My queue says: Ready: 4 Unacked: 0 Total: 4 - is that correct?

**Comment by SCCMOG** [comment]
No bud your q should be empty if you have run the consuming script/service. That script service also turn the lights on and off. Want me to remote in and have a look this evening?

**Comment by RevillWeb** [comment]
Let me try and save you some time first :) Now I know that the queue should be empty then obviously the service isn't able to read the q and deal with them. Let me see if I can figure out why that's happening 👍

**Comment by RevillWeb** [comment]
I found the issue, in `services/LightwaveRF_service.py` I had to change line 25 (`i = body.split('|', 1)`) to `i = body.decode().split('|', 1)` because before it was throwing the error: `TypeError: a bytes-like object is required, not 'str'`.

Now it appears to be working! Thank you for all your help. I'm going to attempt to roll this out to my main server. I'd still be interested in getting the sockets working, I might even be able to see if I can do it myself. If I do get it working I'll create a pull request for you to review.

**Comment by SCCMOG** [comment]
Happy days!
This must be because of the OS version you are on, I doubt you will see it on Hassbian.
What are you running in your prod ENV?

**Comment by RevillWeb** [comment]
Prod env is Ubuntu Server 17.10. Just got it set-up and working nicely. Thinking about this I've realised home much I rely on the Lightwave RF switches! So I'd better figure out how to integrate them :)
